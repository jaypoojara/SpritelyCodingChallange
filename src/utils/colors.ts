export default {
  primary: '#003A67',
  secondary: '#759FC0',
  darkYellow: '#FFCD0C',
  white: '#ffffff',
  black: '#000000',
  textColor: '#484848',
  darkRed: '#C36165',
  lightRed: '#D88789',
  darkBrownYellow: '#C39661',
  lightBrownYellow: '#C39661',
  darkAquamarine: '#62C3B4',
  lightAquamarine: '#88D9CC',
  darkGreen: '#62C36F',
  lightGreen: '#87D893',
  darkPurple: '#6962C3',
  lightPurple: '#8E88D8',
};
