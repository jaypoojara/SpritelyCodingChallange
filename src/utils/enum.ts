export enum TYPE {
  NOTICEBOARD = 'noticeboard',
  ACTIVITIES = 'activities',
  CALENDAR = 'calendar',
  MAINTENANCE = 'maintenance',
  SETTINGS = 'settings',
}
