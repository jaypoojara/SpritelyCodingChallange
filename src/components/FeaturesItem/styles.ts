import {StyleSheet} from 'react-native';
import colors from '../../utils/colors';

const styles = StyleSheet.create({
  itemGradientView: {
    height: 60,
    width: 45,
    borderTopLeftRadius: 8,
    borderBottomLeftRadius: 8,
    justifyContent: 'center',
    alignItems: 'center',
  },
  itemText: {
    color: colors.textColor,
    fontWeight: 'bold',
    fontSize: 20,
    letterSpacing: 0.8,
    paddingHorizontal: 20,
  },
  itemView: {
    backgroundColor: colors.white,
    elevation: 5,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0.3,
    shadowRadius: 6,
    marginBottom: 20,
    borderRadius: 8,
    flexDirection: 'row',
    alignItems: 'center',
  },
});

export default styles;
