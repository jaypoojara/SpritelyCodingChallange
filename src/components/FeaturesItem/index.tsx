import { Text, TouchableOpacity } from 'react-native';
import React from 'react';
import LinearGradient from 'react-native-linear-gradient';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import { TYPE } from '../../utils/enum';
import colors from '../../utils/colors';
import styles from './styles';

interface IProps {
  name: string;
  onPress?(name: string): void;
}

export default function FeaturesItem({ name, onPress }: IProps) {
  const getItemIcon = (type: string) => {
    switch (type) {
      case TYPE.NOTICEBOARD:
        return (
          <LinearGradient
            testID={name}
            colors={[colors.darkPurple, colors.lightPurple]}
            style={styles.itemGradientView}>
            <Ionicons
              name="md-newspaper-outline"
              size={30}
              color={colors.white}
            />
          </LinearGradient>
        );

      case TYPE.ACTIVITIES:
        return (
          <LinearGradient
            testID={name}
            colors={[colors.darkGreen, colors.lightGreen]}
            style={styles.itemGradientView}>
            <MaterialIcons name="directions-walk" size={30} color={colors.white} />
          </LinearGradient>
        );
      case TYPE.CALENDAR:
        return (
          <LinearGradient
            testID={name}
            colors={[colors.darkAquamarine, colors.lightAquamarine]}
            style={styles.itemGradientView}>
            <MaterialIcons
              name="calendar-today"
              size={30}
              color={colors.white}
            />
          </LinearGradient>
        );
      case TYPE.MAINTENANCE:
        return (
          <LinearGradient
            testID={name}
            colors={[colors.darkBrownYellow, colors.lightBrownYellow]}
            style={styles.itemGradientView}>
            <Ionicons name="md-hammer-sharp" size={30} color={colors.white} />
          </LinearGradient>
        );
      case TYPE.SETTINGS:
        return (
          <LinearGradient
            testID={name}
            colors={[colors.darkRed, colors.lightRed]}
            style={styles.itemGradientView}>
            <Ionicons
              name="md-settings-outline"
              size={30}
              color={colors.white}
            />
          </LinearGradient>
        );

      default:
        return null;
    }
  };

  const stringFormat = (string: string) => {
    return string?.charAt(0).toUpperCase() + string?.slice(1);
  };

  return (
    <TouchableOpacity
      testID='features-item-id'
      onPress={() => onPress && onPress(name)}
      style={styles.itemView}>
      {getItemIcon(name)}
      <Text style={styles.itemText}>{stringFormat(name)}</Text>
    </TouchableOpacity>
  );
}
