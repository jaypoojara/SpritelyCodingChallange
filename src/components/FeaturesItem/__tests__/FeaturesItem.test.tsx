import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react-native';
import FeaturesItem from '../index';
import {TYPE} from '../../../utils/enum';

describe('<FeaturesItem />', () => {
  it('renders correctly', () => {
    const element = renderer.create(<FeaturesItem name="" />).toJSON();
    expect(element).toMatchSnapshot();
  });

  it('should render the expected text: Noticeboard', () => {
    render(<FeaturesItem name={TYPE.NOTICEBOARD} />);
    const find = screen.getByText('Noticeboard');
    expect(find).toBeTruthy();
  });

  it('should render the expected text: Activities', () => {
    render(<FeaturesItem name={TYPE.ACTIVITIES} />);
    const find = screen.getByText('Activities');
    expect(find).toBeTruthy();
  });

  it('should render the expected text: Calendar', () => {
    render(<FeaturesItem name={TYPE.CALENDAR} />);
    const find = screen.getByText('Calendar');
    expect(find).toBeTruthy();
  });

  it('should render the expected text: Maintenance', () => {
    render(<FeaturesItem name={TYPE.MAINTENANCE} />);
    const find = screen.getByText('Maintenance');
    expect(find).toBeTruthy();
  });

  it('should render the expected text: Settings', () => {
    render(<FeaturesItem name={TYPE.SETTINGS} />);
    const find = screen.getByText('Settings');
    expect(find).toBeTruthy();
  });

  it('noticeboard icon render', () => {
    render(<FeaturesItem name={TYPE.NOTICEBOARD} />);
    const element = screen.queryByTestId(TYPE.NOTICEBOARD);
    expect(element).toBeTruthy();
  });

  it('activities icon render', async () => {
    render(<FeaturesItem name={TYPE.ACTIVITIES} />);
    const element = screen.queryByTestId(TYPE.ACTIVITIES);
    expect(element).toBeTruthy();
  });

  it('calendar icon render', () => {
    render(<FeaturesItem name={TYPE.CALENDAR} />);
    const element = screen.getByTestId(TYPE.CALENDAR);
    expect(element).toBeTruthy();
  });

  it('maintenance icon render', () => {
    render(<FeaturesItem name={TYPE.MAINTENANCE} />);
    const element = screen.getByTestId(TYPE.MAINTENANCE);
    expect(element).toBeTruthy();
  });

  it('settings icon render', () => {
    render(<FeaturesItem name={TYPE.SETTINGS} />);
    const element = screen.getByTestId(TYPE.SETTINGS);
    expect(element).toBeTruthy();
  });
});
