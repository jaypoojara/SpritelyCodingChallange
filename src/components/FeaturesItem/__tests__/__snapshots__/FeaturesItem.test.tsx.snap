// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturesItem /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 8,
      "elevation": 5,
      "flexDirection": "row",
      "marginBottom": 20,
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 6,
    }
  }
  testID="features-item-id"
>
  <Text
    style={
      Object {
        "color": "#484848",
        "fontSize": 20,
        "fontWeight": "bold",
        "letterSpacing": 0.8,
        "paddingHorizontal": 20,
      }
    }
  />
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 8,
      "elevation": 5,
      "flexDirection": "row",
      "marginBottom": 20,
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 6,
    }
  }
  testID="features-item-id"
>
  <Text
    style={
      Object {
        "color": "#484848",
        "fontSize": 20,
        "fontWeight": "bold",
        "letterSpacing": 0.8,
        "paddingHorizontal": 20,
      }
    }
  />
</View>
`;
